storage/src/storage.rs

```diff
@@ -160,6 +163,9 @@ pub trait Storage: Send + Sync {
 
     /// Similar to `Storage::get_canon_blocks`, gets hashes of all blocks subject to `filter` and `limit` in block-number ascending order. A maintenance function, not intended for general use.
     async fn get_block_hashes(&self, limit: Option<u32>, filter: BlockFilter) -> Result<Vec<Digest>>;
+
+    /// Performs low-level storage validation; it's mostly intended for test purposes, as there is a lower level `KeyValueStorage` interface available outside of them.
+    async fn validate(&self, limit: Option<u32>, fix_mode: FixMode) -> bool;
```

Comment:
we should make the docs here generic as to future implementations, which would presumably validate themselves as well -- maybe in a strongly typed relational DB that would be a nop, but thats okay.